gather_istio.sh

```diff
@@ -0,0 +1,243 @@
+  ns=$2
+
+  echo
+  if [ -n "$ns" ]; then
+    echo "Inspecting resource ${resource} in namespace ${ns}"
+    # it's here just to make the linter happy (we have to use double quotes arround the variable)
+    if [ -n "${log_collection_args}" ]
+    then
+      oc adm inspect "${log_collection_args}" "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+    else
+      oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+    fi
+  else
+    echo "Inspecting resource ${resource}"
+    # it's here just to make the linter happy
+    if [ -n "${log_collection_args}" ]
+    then
+      oc adm inspect "${log_collection_args}" "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+    else
+      oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+    fi
+  fi
+}
+
+function inspectNamespace() {
+  local ns
+  ns=$1
+
+  inspect "ns/$ns"
+  for crd in $crds; do
+    inspect "$crd" "$ns"
+  done
+  inspect net-attach-def,roles,rolebindings "$ns"
+}
+
+function main() {
+  local crds
+  echo
+  echo "Executing Istio gather script"
+  echo
+
+  versionFile="${BASE_COLLECTION_PATH}/version"
+  echo "openshift-service-mesh/must-gather"> "$versionFile"
+  version >> "$versionFile"
+
+  # set global variable which is used when calling 'oc adm inspect'
+  get_log_collection_args
+
+  operatorNamespace=$(oc get pods --all-namespaces -l app.kubernetes.io/created-by=servicemesh-operator3 -o jsonpath="{.items[0].metadata.namespace}")
+  # this gets also logs for all pods in that namespace
+  inspect "ns/$operatorNamespace"
+  inspect clusterserviceversion "${operatorNamespace}"
+
+  inspect nodes
+
+  # TODO: we have to add a new label in the helm postrenderer section in the operator the same way like 'metadata.ownerReferences' is added
+  # 'install.operator.istio.io/owning-resource' should not be used as it could interfere with istioctl
+  # using 'metadata.ownerReferences' is not a good solution, we should use the new label when ready
+  # this needs to be revisited when working on https://issues.redhat.com/browse/OSSM-6804
+  for r in $(oc get clusterroles,clusterrolebindings -l install.operator.istio.io/owning-resource -oname); do
+    inspect "$r"
+  done
+  for r in $(oc get clusterroles,clusterrolebindings -l 'app in (istiod,istio-reader)' -oname); do
+    inspect "$r"
+  done
+
+  # inspect all istio.io CRDs
+  crds="$(getCRDs)"
+  for crd in ${crds}; do
+    inspect "crd/${crd}"
+  done
+
+  # inspect all controlled mutatingwebhookconfiguration
+  for mwc in $(oc get mutatingwebhookconfiguration -l app=sidecar-injector -o name);do
+    inspect "${mwc}"
+  done
+  # inspect all controlled validatingwebhookconfiguration
+  for vwc in $(oc get validatingwebhookconfiguration -l app=istiod -o name);do
+    inspect "${vwc}"
+  done
+
+  # this will just store all CRs as those are cluster-scoped resources
+  inspect "Istio"
+  inspect "IstioRevision"
+  inspect "IstioCNI"
+  #TODO: remoteIstio
+  #TODO: kiali??
```

Comment:
do we have tickets for these two? if not let's create them

Reply:
Yes, it will be covered by https://issues.redhat.com/browse/OSSM-6804